detox/src/DetoxExportWrapper.js

```diff
@@ -35,7 +34,18 @@ class DetoxExportWrapper {
       Detox.none.initContext(global);
     }
 
-    this[_detox] = await DetoxExportWrapper._initializeInstance(config, params);
+    Detox.none.setError(null);
+
+    try {
+      this[_detox] = DetoxExportWrapper._createInstance(config);
+      await this[_detox].init(params);
```

Comment:
It took a while for me to understand why this is indeed necessary but I got there eventually 😄 
Maybe adding some notes here would help for future reference! As you mentioned, this change is very important and its effect is two-fold (timeouts - handle screenshots, jasmine hack).

Reply:
Could we manifest this behavior using an E2E test?

Reply:
Added minimal e2e tests for Jasmine and Circus in the last commits, though they don't specifically cover 2 from the point of the assertions.